Raft cli docs (#8548)
hashicorp/vault: website/pages/docs/commands/operator/raft.mdx

---
layout: docs
page_title: operator raft - Command
sidebar_title: <code>raft</code>
description: >-
  The "operator raft" command is used to interact with the integrated Raft storage backend.
---

# operator raft

> This command groups subcommands for operators to manage the integrated raft storage backend.

Reviewer: ```suggestion
This command groups subcommands for operators to manage the integrated Raft storage backend.
```
We should be consistent with capitalization - it seems like Consul mostly uses 'Raft' over 'raft', so can we stick with that?

Author: Sure. Done.

```text
Usage: vault operator raft <subcommand> [options] [args]

 This command groups subcommands for operators interacting with the Vault raft
 storage backend. Most users will not need to interact with these commands. Here
 are a few examples of the raft operator commands:

Subcommands:
    join           Joins a node to the raft cluster
    list-peers     Returns the raft peer set
    remove-peer    Removes a node from the raft cluster
    snapshot       Restores and saves snapshots from the raft cluster
```

## join

> This command is used to join a new node as a peer to the raft cluster.

Reviewer: ```suggestion
This command is used to join a new node as a peer to the Raft cluster. In order to join, there must be at least one existing member of the cluster. Agents can join other agents multiple times without issue. If a node that is already part of a cluster joins another node, then the clusters of the two nodes join to become a single cluster.
```
I think it would be nice to include more information generally.

Author: Some of the things in ☝️ are applicable only to Consul. Took the first sentence though. Thanks!

          "protocol_version": "3",
          "voter": true
        }
      ]
    }
  }
}
```

## remove-peer

This command is used to remove a node from being a peer to the raft cluster. In
certain cases where a peer may be left behind in the Raft configuration even
though the server is no longer present and known to the cluster, this command
can be used to remove the failed server so that it is no longer affects the Raft
quorum.

```text
Usage: vault operator raft remove-peer <server_id>

  Removes a node from the raft cluster.

	  $ vault operator raft remove-peer node1
```

## snapshot

This command groups subcommands for operators interacting with the snapshot
functionality of the raft storage backend. There are 2 subcommands supported:
`save` and `restore`.

```text
Usage: vault operator raft snapshot <subcommand> [options] [args]

  This command groups subcommands for operators interacting with the snapshot
  functionality of the raft storage backend.

Subcommands:
    restore    Installs the provided snapshot, returning the cluster to the state defined in it
    save       Saves a snapshot of the current state of the raft cluster into a file
```

### snapshot save

> Takes a snapshot of the raft cluster state.

Reviewer: Is this just the Raft cluster state, or is it the entirety of the storage backend? I think we need to clarify.

Author: It takes backup of the storage. Clarified.

---
layout: docs
page_title: operator raft - Command
sidebar_title: <code>raft</code>
description: >-
  The "operator raft" command is used to interact with the integrated Raft storage backend.
---

# operator raft

This command groups subcommands for operators to manage the integrated Raft storage backend.

```text
Usage: vault operator raft <subcommand> [options] [args]

 This command groups subcommands for operators interacting with the Vault
 integrated Raft storage backend. Most users will not need to interact with these
 commands. Here are a few examples of the Raft operator commands:

Subcommands:
    join           Joins a node to the Raft cluster
    list-peers     Returns the Raft peer set
    remove-peer    Removes a node from the Raft cluster
    snapshot       Restores and saves snapshots from the Raft cluster
```

## join

This command is used to join a new node as a peer to the Raft cluster. In order

> to join, there must be at least one existing member of the cluster.

Reviewer 2: Can you mention that if using shamir as your unseal mechanism the node will need to be unsealed before it can fully join the cluster?

quorum.

```text
Usage: vault operator raft remove-peer <server_id>

  Removes a node from the Raft cluster.

	  $ vault operator raft remove-peer node1
```

## snapshot

This command groups subcommands for operators interacting with the snapshot
functionality of the integrated Raft storage backend. There are 2 subcommands
supported: `save` and `restore`.

```text
Usage: vault operator raft snapshot <subcommand> [options] [args]

  This command groups subcommands for operators interacting with the snapshot
  functionality of the integrated Raft storage backend.

Subcommands:
    restore    Installs the provided snapshot, returning the cluster to the state defined in it
    save       Saves a snapshot of the current state of the Raft cluster into a file
```

### snapshot save

Takes a snapshot of the Vault data. The snapshot can be used to restore Vault to
the point in time when a snapshot was taken.

```text
Usage: vault operator raft snapshot save <snapshot_file>

  Saves a snapshot of the current state of the Raft cluster into a file.

	  $ vault operator raft snapshot save raft.snap
```

### snapshot restore

> Restores a snapshot of the Raft cluster state.

Reviewer: I am being kind of nitpicky only because this is publicly facing but -

```suggestion
Restores a snapshot of Vault data taken with `vault operator raft snapshot save`.
```
